jdaviz/configs/imviz/plugins/tools.py

```diff
@@ -84,3 +86,61 @@ def on_mouse_or_key_event(self, data):
                     viewer.state.x_max = x + dx / 2
                     viewer.state.y_min = y - dy / 2
                     viewer.state.y_max = y + dy / 2
+
+
+@viewer_tool
+class BqplotContrastBias(CheckableTool):
+
+    icon = 'glue_contrast'
+    tool_id = 'bqplot:contrastbias'
+    action_text = 'Adjust contrast/bias'
+    tool_tip = 'Click and drag to adjust, double-click to reset'
+
+    def __init__(self, viewer, **kwargs):
+        self._time_last = 0
+        super().__init__(viewer, **kwargs)
+
+    def activate(self):
+        self.viewer.add_event_callback(self.on_mouse_or_key_event,
+                                       events=['dragstart', 'dragmove',
+                                               'dragend', 'dblclick'])
+
+    def deactivate(self):
+        self.viewer.remove_event_callback(self.on_mouse_or_key_event)
+
+    def on_mouse_or_key_event(self, data):
+
+        event = data['event']
+
+        # Note that we throttle this to 200ms here as changing the contrast
+        # and bias it expensive since it forces the whole image to be redrawn
+        if event == 'dragmove':
+            if (time.time() - self._time_last) <= 0.2:
+                return
+
+            event_x = data['domain']['x']
+            event_y = data['domain']['y']
+
+            if ((event_x < self.viewer.state.x_min) or
+                    (event_x >= self.viewer.state.x_max) or
+                    (event_y < self.viewer.state.y_min) or
+                    (event_y >= self.viewer.state.y_max)):
+                return
+
+            x = event_x / (self.viewer.state.x_max - self.viewer.state.x_min)
+            y = event_y / (self.viewer.state.y_max - self.viewer.state.y_min)
+            state = self.viewer.layers[0].state
```

Comment:
Actually, hold on... Now that we have blinking. When I blink to the second image, this tool still is locked onto the first one. We cannot hard-code `self.viewer.layers[0].state` here. @astrofrog , what is the proper way to grab the Data layer that is being displayed?

Reply:
Or maybe @maartenbreddels knows? 🙏

Reply:
I think it's beging displayed in order, so you could do sth like
`layer_top = [l for l in self.viewer.layers if l.visible][-1]`
Does that help?

Reply:
Yes, thanks!